version string 5.0.0-drop04.0 cannot be parsed

Version: 1.8.3

when started spring boot application against a neo4j 5.0.0-drop04.0 version, the startup fails with the following error.

```
2022-07-27 15:58:35.928 ERROR 6776 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'neo4jMigrationsInitializer' defined in class path resource [ac/simons/neo4j/migrations/springframework/boot/autoconfigure/MigrationsAutoConfiguration.class]: Invocation of init method failed; nested exception is java.lang.NumberFormatException: For input string: "0-drop04"
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[spring-context-5.3.21.jar:5.3.21]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.21.jar:5.3.21]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.6.9.jar:2.6.9]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:745) ~[spring-boot-2.6.9.jar:2.6.9]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:420) ~[spring-boot-2.6.9.jar:2.6.9]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-2.6.9.jar:2.6.9]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1317) ~[spring-boot-2.6.9.jar:2.6.9]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306) ~[spring-boot-2.6.9.jar:2.6.9]
	at com.neo4j.opsmanager.server.ServerWebApplication.main(ServerWebApplication.java:19) ~[classes/:na]
Caused by: java.lang.NumberFormatException: For input string: "0-drop04"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67) ~[na:na]
	at java.base/java.lang.Integer.parseInt(Integer.java:668) ~[na:na]
	at java.base/java.lang.Integer.valueOf(Integer.java:999) ~[na:na]
	at ac.simons.neo4j.migrations.core.internal.Neo4jVersionComparator.compare(Neo4jVersionComparator.java:52) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.VersionPrecondition.lambda$isMet$1(VersionPrecondition.java:114) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at java.base/java.util.Optional.map(Optional.java:260) ~[na:na]
	at ac.simons.neo4j.migrations.core.VersionPrecondition.isMet(VersionPrecondition.java:113) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.DiscoveryService.lambda$hasUnmetPreconditions$6(DiscoveryService.java:127) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[na:na]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[na:na]
	at ac.simons.neo4j.migrations.core.DiscoveryService.hasUnmetPreconditions(DiscoveryService.java:127) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.DiscoveryService.lambda$findMigrations$0(DiscoveryService.java:73) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at java.base/java.util.ArrayList.removeIf(ArrayList.java:1672) ~[na:na]
	at java.base/java.util.ArrayList.removeIf(ArrayList.java:1660) ~[na:na]
	at ac.simons.neo4j.migrations.core.DiscoveryService.findMigrations(DiscoveryService.java:73) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.Migrations.getMigrations(Migrations.java:102) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.Migrations.lambda$apply$2(Migrations.java:177) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.Migrations.executeWithinLock(Migrations.java:369) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.core.Migrations.apply(Migrations.java:176) ~[neo4j-migrations-1.8.3.jar:1.8.3]
	at ac.simons.neo4j.migrations.springframework.boot.autoconfigure.MigrationsInitializer.afterPropertiesSet(MigrationsInitializer.java:45) ~[neo4j-migrations-spring-boot-autoconfigure-1.8.3.jar:1.8.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863) ~[spring-beans-5.3.21.jar:5.3.21]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800) ~[spring-beans-5.3.21.jar:5.3.21]
	... 16 common frames omitted
```

--- a/neo4j-migrations-core/src/main/java/ac/simons/neo4j/migrations/core/internal/Neo4jVersionComparator.java
+++ b/neo4j-migrations-core/src/main/java/ac/simons/neo4j/migrations/core/internal/Neo4jVersionComparator.java
@@ -29,7 +29,7 @@ public final class Neo4jVersionComparator implements Comparator<String> {
 	static String[] sanitize(String in) {
 		return in
 			.replaceFirst("(?i)^Neo4j/", "")
-			.replaceFirst("-\\w+$", "")
+			.replaceFirst("-[\\w.]+$", "")
 			.split("\\.");
 	}
 

--- a/neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/internal/Neo4jVersionComparatorTest.java
+++ b/neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/internal/Neo4jVersionComparatorTest.java
@@ -37,6 +37,7 @@ class Neo4jVersionComparatorTest {
 		"4, 4.0.0",
 		"Neo4j/4.4-aura, 4.4",
 		"4.4-aura, 4.4",
+		"5.0.0-drop04.0, 5.0"
 	})
 	void equalsShouldWork(String v1, String v2) {
 		Assertions.assertThat(VERSION_COMPARATOR.compare(v1, v2)).isZero();

